ReactSkia/views/common/RSkImageCacheManager.cpp

```diff
@@ -20,121 +21,91 @@ using namespace std;
+  return (evictCount == EVICT_COUNT);
+}
 
-}//namespace
-
-void configure() {
+void RSkImageCacheManager::init() {
+  RSkImageCacheManager &cacheInstance= RSkImageCacheManager::getImageCacheManagerInstance();
   SkGraphics::SetResourceCacheTotalByteLimit(SKIA_CPU_IMAGE_CACHE_LIMIT);
 #ifdef RNS_SHELL_HAS_GPU_SUPPORT
   GrDirectContext* gpuContext = RSkSurfaceWindow::getDirectContext();
-    if(gpuContext)
-      gpuContext->setResourceCacheLimit(SKIA_GPU_IMAGE_CACHE_LIMIT);
-#endif /*RNS_SHELL_HAS_GPU_SUPPORT*/
-}
-
-sk_sp<SkImage> getImageData(const char *path) {
-
-  sk_sp<SkImage> imageData{nullptr};
-
-  if(!path) {
-    RNS_LOG_ERROR("Invalid File");
-    return nullptr;
-  }
-
-  std::scoped_lock lock(mutexLock);
-
- /*First to check file entry presence . If not exist, generate imageData*/
-  ImageCacheMap::iterator it = imageCache.find(path);
-  imageData= ((it != imageCache.end()) ? it->second : nullptr);
-  if(!imageData) {
-    imageData = makeImageData(path);
-    /*Add entry to hash map only if the cache mem usage is with in the limit*/
-    /* TO DO: Eviction mechanism to be improved by decouple it from getImageData,
-              through worker thread ....*/
-    if(evictAsNeeded() && imageData) {
-      imageCache.insert(pair<string, sk_sp<SkImage>>(path,imageData));
-      RNS_LOG_DEBUG("New Entry in Map..."<<" file :"<<path);
-    } else {
-        RNS_LOG_DEBUG("Cache Memory limit reached or Couldn't create imageData, So file not cached ...");
-    }
-
-  }
-  return imageData;
+  if(gpuContext)
+    gpuContext->setResourceCacheLimit(SKIA_GPU_IMAGE_CACHE_LIMIT);
+#endif //RNS_SHELL_HAS_GPU_SUPPORT
 }
 
 #ifdef RNS_IMAGE_CACHE_USAGE_DEBUG
 void printCacheUsage() {
   static size_t prevCpuUsedMem{0},prevGpuUsedMem{0};
   size_t usageArr[2]={0,0};
-
-  getCacheUsage(usageArr);
+  RSkImageCacheManager::getCacheUsage(usageArr);
   RNS_LOG_INFO("Memory consumed for this run in CPU CACHE :"<<(usageArr[CPU_MEM_ARR_INDEX] - prevCpuUsedMem));
   prevCpuUsedMem = usageArr[CPU_MEM_ARR_INDEX];
 #ifdef RNS_SHELL_HAS_GPU_SUPPORT
-    RNS_LOG_INFO("Memory consumed for this run in GPU CACHE:"<<(usageArr[GPU_MEM_ARR_INDEX] - prevGpuUsedMem));
-    prevGpuUsedMem = usageArr[GPU_MEM_ARR_INDEX];
+  RNS_LOG_INFO("Memory consumed for this run in GPU CACHE:"<<(usageArr[GPU_MEM_ARR_INDEX] - prevGpuUsedMem));
+  prevGpuUsedMem = usageArr[GPU_MEM_ARR_INDEX];
 #endif
 }
-#endif/*RNS_IMAGE_CACHE_USAGE_DEBUG*/
+#endif//RNS_IMAGE_CACHE_USAGE_DEBUG
 
-} //RSkImageCacheManager
+sk_sp<SkImage> RSkImageCacheManager::findImageDataInCache(const char* path) {
+  sk_sp<SkImage> imageData{nullptr};
+  ImageCacheMap::iterator it = imageCache_.find(path);
+  imageData= ((it != imageCache_.end()) ? it->second : nullptr);
+  return imageData;
+}
+void RSkImageCacheManager::imageDataInsertInCche(const char* path,sk_sp<SkImage> imageData){
+  imageCache_.insert(std::pair<std::string, sk_sp<SkImage>>(path,imageData));
```

Comment:
in this function , the implementation need to be like below 
If (imageData && evictAsNeeded()) {
   insert()
}

Reply:
also please add logging messages which was there in the old code for this logic .